x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -998,11 +1012,17 @@ public LogicalPlan apply(LogicalPlan plan) {
                     recheck = false;
                     if (p instanceof Aggregate aggregate) {
                         var remaining = seenProjection.get() ? removeUnused(aggregate.aggregates(), used) : null;
-                        // no aggregates, no need
+
                         if (remaining != null) {
                             if (remaining.isEmpty()) {
-                                recheck = true;
-                                p = aggregate.child();
+                                // We still need to have a plan that produces 1 row per group.
+                                if (aggregate.groupings().isEmpty()) {
+                                    p = emptyRow(aggregate.source());
+                                } else {
+                                    // Aggs cannot produce pages with 0 columns, so retain one grouping.
+                                    remaining = List.of((NamedExpression) aggregate.groupings().get(0));
```

Comment:
Instead of casting, use Expressions.wrapAsNamed or even better make a reference to the group (to avoid copying the alias and its nested expression): Expressions.attribute()